v4 Redo List of Log Names

Ade2310 opened the pull request:
## Description

Filled out the lists of things that can be logged, and cleaned up other things on this page. 

## Checklist

* [√] Markdown formatting has been checked and renders correctly with no issue
* [√] All information provided in the pull request is accurate and up to date with MoonlightBot's functioning
* [√ ] All information provided is consistent with the already existing style of the rest of documentation (skip this if you're modifying the style as whole)
* [ √] Reasonable efforts have been made to make the documentation readable to the end user
* [√ ] Reasonable effort has been put in proofreading the text for any spelling, typography or grammar errors
* [ √] If this pull request features changes that have been made in beta versions, it is immediate to the end user to understand this



# ALSO DID YOU KNOW CARBON MONOXIDE POISONING CANONICALLY EXISTS IN SOFIA THE FIRST

Ade2310 committed:
Filled out the lists of things that can be logged, and cleaned up other things on this page.

* [√] Markdown formatting has been checked and renders correctly with no issue
* [√] All information provided in the pull request is accurate and up to date with MoonlightBot's functioning
* [√ ] All information provided is consistent with the already existing style of the rest of documentation (skip this if you're modifying the style as whole)
* [ √] Reasonable efforts have been made to make the documentation readable to the end user
* [√ ] Reasonable effort has been put in proofreading the text for any spelling, typography or grammar errors
* [ √] If this pull request features changes that have been made in beta versions, it is immediate to the end user to understand this

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
 # List of log names
 
-## Categories
+Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names(which can be used in place of individual log names, see [configuring channels](../admin-commands/config/configuring-channels.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an as ...
```
1. No space names(which
2. "configuring channels" page link here is incorrect and points to the old format
3. Too little emphasis on the config link anyways, think of someone landing for the first time on this page and thinking "okay the list, what do I do with this?"

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
 
 {% hint style="warning" %}
-Log names are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names are _Capitalized_
+Some logs require the bot to have the "View Audit Logs" permission enabled.
```
Audit Log is singular on Discord

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
-{% endhint %}
+## Moderation
+
+This category is used for actions by moderators or ones that moderators could use. Most of them are fired when a command is run unless they start with AUDIT_. In that case, actions that are done with or without using the bot are logged.
```
Explains the audit values are taken from the server's audit log

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+
+| Name              | Emitted when     |
+| ----------------- | ---------------- |
+| BAN               | Someone gets banned |
```
"someone" in much of these cases is too generic, try using more accurate terms like "A user". Use passive voice too

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+| INFRACTION_DELETE | Someone removes an infraction|
+| KICK              | Someone gets kicked |
+| MUTE              | Someone gets muted |
+| PAUSE_ROLE_ADD    | An expiring role gets added  |
```
"Expiring role"? No, these are for the `/pause-role` command which temporarily removes a role and adds it back at the expiration

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+| REASON_UPDATE     | A reason for an infraction gets updated |
+| ROLE_ADD          | Someone adds a role |
+| ROLE_REMOVE       | Someone removes a role|
+| SELFROLE_ADD      | Someone adds a user-assignable role |
```
self-assignable role, link to the relevant command page

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+
+| Name                | Emitted when     |
+| ------------------- | ---------------- |
+| CHANNEL_CLEAN       | Someone cleans all the messages from a channel |
```
Specify it contains a file with the deleted messages

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+| ------------------- | ---------------- |
+| CHANNEL_CLEAN       | Someone cleans all the messages from a channel |
+| MESSAGE_BULK_DELETE | Someone bulk deletes a set of messages |
+| MESSAGE_DELETE      | Anyone deletes a message |
```
...A message is deleted, passive voice

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
-| REACTION\_ROLE_\__DEBUG | Debug for reaction roles related actions |
+| Name                | Emitted when     |
+| ------------------- | ---------------- |
+| REACTION_ROLE_DEBUG | Someone uses a reaction role |
```
Incorrect, this one is for when someone triggers a reaction role but something goes wrong. It's meant to help the server admins understand what's wring with the server's reaction role configuration.

MoonlightCapital reviewed (changes requested):
Overall good, some words are too generic and it would be nice if most of the log descriptions associated to a command linked back to the command's specific page or a tutorial where applicable. Only the first for each command preferably.

Ade2310 committed:
Hopefully better than the first time
Fixed all the mentioned things

"I am not Dad, I am Magic Claw. Magic Claw has no children. His days are free and easy!" - Bandit"

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -4,43 +4,43 @@ description: On this page all the available logs are listed
 # List of log names
 
-Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names(which can be used in place of individual log names, see [configuring channels](../admin-commands/config/configuring-channels.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an as ...
+Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to ...
```
Link this to the # channels anchor to bring the reader to the correct session upon clicking

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -4,43 +4,43 @@ description: On this page all the available logs are listed
 ## Moderation
 
-This category is used for actions by moderators or ones that moderators could use. Most of them are fired when a command is run unless they start with AUDIT_. In that case, actions that are done with or without using the bot are logged.
+This category is used for actions by moderators or ones that moderators could use. Most of them are fired when a command is run unless they start with AUDIT_, which uses the values in the server's audit log. In that case, actions that are done with or without using the bot are logged.
```
Wording here is convoluted:

1. You don't "use" an action
2. Data is read from the audit log, "uses" is too generic"
3. "with or without" sounds repetitive

Suggested rewrite:
> This category is to log actions taken by moderators. Most of them are triggered by using bot commands, but those starting with \`AUDIT_\` will read the data from Discord's audit log providing MoonlightBot has permission to do so.

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -4,43 +4,43 @@ description: On this page all the available logs are listed
-| PAUSE_ROLE_EXTEND | A role expiration date gets extended |
-| PAUSE_ROLE_REMOVE | A role gets temporarily removed |
+| BAN               | A user gets banned |
+| INFRACTION_DELETE | A moderator removes an infraction|
```
Deletes is more accurate than removes, links the pages to the relevant (sub)commands to take the actions.

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -4,43 +4,43 @@ description: On this page all the available logs are listed
-| TEMPMUTE_REMOVE   | Someone's temporary ban gets removed |
+| ROLE_ADD          | A role gets added |
+| ROLE_REMOVE       | A role gets removed|
+| SELFROLE_ADD      | A user adds a self-assignable role (see [Selfrole](../role-management-commands/selfrole.md)) |
```
I prefer the Wikipedia style of linking directly in the words without adding content to make them clickable.

E.g. `A user adds a [self-assignable role](../role-management-commands/selfrole.md)`

And conclude the sentence with "to themselves".

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -57,10 +57,10 @@ This category is for things having to do with messages.
-| MESSAGE_BULK_DELETE | Someone bulk deletes a set of messages |
-| MESSAGE_DELETE      | Anyone deletes a message |
-| MESSAGE_EDIT        | Anyone edits a message |
+| CHANNEL_CLEAN       | A moderator cleans all the messages from a channel (which creates a file with the deleted messages) |
```
Use passive voice, parenthesis is unsound

`Messages get cleaned from a channel. Includes a file with a backup of the deleted messages`

And the link to the relevant page on "cleaned"

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -73,4 +73,4 @@ This category is for debugging purposes. Logs in this category are meant to aid
 | Name                | Emitted when     |
 | ------------------- | ---------------- |
-| REACTION_ROLE_DEBUG | Someone uses a reaction role |
+| REACTION_ROLE_DEBUG | A user triggers a reaction role, and something goes wtong |
```
Too vague and there's a typo. This log is triggered, say, when permissions are misconfigured or someone tries to get a join-only role or similar things

MoonlightCapital reviewed (changes requested):
I forgot to say that all logs require the bot to have permission to view, send messages, embeds, and files in the log channel, this is 100% my bad and should be added in the intro

Ade2310 committed:
Big thing: Hyperlinks
Also fixed everything else called out

Fun Fact: Unicourse loves eating ~~children~~ i mean chicken bucket

Ade2310 committed:
Added the sentence about how it needs the permissions

Also PAW PATROL CANONICALY EXISTS IN BLUEY BUT IT"S CATS NOT DOGS

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
 ---
 
-# List of log names
+# [List of log names](list-of-log-names.md#list-of-log-names)
```
```suggestion
# List of Log Names
```
Remove hyperlink here

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+# [List of log names](list-of-log-names.md#list-of-log-names)
 
-## Categories
+Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to ...
```
The anchor to the channels part goes here after config.md, remove the markdown. I realized now how disconnected the sentence is with the linking to guide part is popping off in the middle.

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
 
 {% hint style="warning" %}
-Log names are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names are _Capitalized_
+Some logs require the bot to have the "View Audit Log" permission enabled.
```
Move all the permission-related part in this box from the paragraph above.

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+| PAUSE_ROLE_ADD    | A role [gets temporarily removed](../role-management-commands/pause-role.md) |
+| PAUSE_ROLE_EXTEND | A role's length of time to be removed gets adjusted |
+| PAUSE_ROLE_REMOVE | A role gets added back |
+| REASON_UPDATE     | A [reason for an infraction](../moderation-commands/infractions.md) gets updated |
```
Link to subcommand # (it's `/infractions reason`)

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+| ROLE_ADD          | A [role](../role-management-commands/role.md) gets added |
+| ROLE_REMOVE       | A role gets removed|
+| SELFROLE_ADD      | A user adds a [self-assignable role](../role-management-commands/selfrole.md) to themselves |
+| SELFROLE_REMOVE   | A user removes a self-assignable role  |
```
```suggestion
| SELFROLE_REMOVE   | A user removes a self-assignable role from themselves  |
```

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+
+## Members
+
+This category is for when someone leaves or joins the server.
```
```suggestion
This category is for actions related to members interacting with the server.
```

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
+| CHANNEL_CLEAN       | Messages get [cleaned](../moderation-commands/clean.md) from a channel. Includes a file with a backup of the deleted messages |
+| MESSAGE_BULK_DELETE | A moderator bulk deletes a set of messages |
+| MESSAGE_DELETE      | A message is deleted |
+| MESSAGE_EDIT        | A user edits a message |
```
```suggestion
| MESSAGE_EDIT        | A message gets edited |
```

Consistency with rest of the table

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,76 @@
-| REACTION\_ROLE_\__DEBUG | Debug for reaction roles related actions |
+| Name                | Emitted when     |
+| ------------------- | ---------------- |
+| REACTION_ROLE_DEBUG | Permissions are misconfigured or someone tries to get a join only role  |
```
Too specific now, my previous comment to this was only an example for you writer. There are more mishaps triggering the RR debug log

MoonlightCapital reviewed (changes requested):
Getting there!

Ade2310 committed:
Hoeefully htis time it's fixed better

TOYOTATHON IS ON LIKE YOUR MOM /j

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -2,13 +2,10 @@
-# [List of log names](list-of-log-names.md#list-of-log-names)
 
-Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to ...
+# [List Of Log Names](../advanced/list-of-log-names.md#ListOfLogNames)
```
No! The header must be the first thing on the page. I did not say to swap these two things, I said remove the unnecessary markdown, links, and add the # to the relevant session after config.md

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -2,13 +2,10 @@
+# [List Of Log Names](../advanced/list-of-log-names.md#ListOfLogNames)
 
-{% hint style="warning" %}
-Some logs require the bot to have the "View Audit Log" permission enabled.
```
Why was this removed? I wanted the `All logs require the bot to have permission to view and send messages in the logging channel. It must also be able to send embeds and files there,and some logs require the bot to have the "View Audit Log" permission enabled.` part to be moved inside here to be consistent with all permissions in one place, rephrased if necessary.

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -73,4 +70,4 @@ This category is for debugging purposes. Logs in this category are meant to aid
 | Name                | Emitted when     |
 | ------------------- | ---------------- |
-| REACTION_ROLE_DEBUG | Permissions are misconfigured or someone tries to get a join only role  |
+| REACTION_ROLE_DEBUG | Someone tries to get a join only role or something else goes wrong in the process of adding a reaction role |
```
The join only role part is only an example that should be removed from the page.

MoonlightCapital reviewed (changes requested):
My previous comments may have been misunderstood. It does not make any sense to have a paragraph before the heading and such heading linking to itself. Markdown on link texts isn't a good idea - let the link formatting suffice.

Ade2310 committed:
... I did something and hopefully fixed it right

ALSO LESBIANS EXIST IN BLUEY

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,9 +1,8 @@
+---
 
-# [List Of Log Names](../advanced/list-of-log-names.md#ListOfLogNames)
+description: On this page all the available logs are listed
```
No, this desc goes on top between the `---` things like before, likely some GitBook specific meaning

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -70,4 +69,4 @@ This category is for debugging purposes. Logs in this category are meant to aid
 | Name                | Emitted when     |
 | ------------------- | ---------------- |
-| REACTION_ROLE_DEBUG | Someone tries to get a join only role or something else goes wrong in the process of adding a reaction role |
+| REACTION_ROLE_DEBUG | Something goes wrong in the process of adding a reaction role |
```
```suggestion
| REACTION_ROLE_DEBUG | An error occurs in the process of triggering a reaction role |
```

This is correctly generalized. Now add the link to the tutorial on the words "reaction role"

MoonlightCapital reviewed (changes requested):
Some slight changes and this is good to go!

Ade2310 committed:
More fixes, hopefully good this time

ALSO: NEW BLUEY EPISODES AAAAAA

Ade2310 committed:
NOW ACTUALLY RIGHT

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,74 @@
 ---
 
-# List of log names
+# List Of Log Names
```
```suggestion
# List of Log Names
```

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,74 @@
-| REACTION\_ROLE_\__DEBUG | Debug for reaction roles related actions |
+| Name                | Emitted when     |
+| ------------------- | ---------------- |
+| REACTION_ROLE_DEBUG | An error occurs in the process of triggering a [reaction role.](/start-up/setting-up-reaction-roles.md) |  
```
```suggestion
| REACTION_ROLE_DEBUG | An error occurs in the process of triggering a [reaction role](/start-up/setting-up-reaction-roles.md) |  
```

MoonlightCapital commented on advanced/list-of-log-names.md:
```diff
@@ -1,72 +1,74 @@
+# List Of Log Names
 
-## Categories
+Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to ...
```
```suggestion
Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [configuring channels](../management-commands/config.md#channels)) are _Capitalized_. Each one of them is separated by categories in the lists below. To log all events, use an asterisk (`*`). All logs require the bot to have permission to view and send messages, embed links and attach files in the respective channels. Some logs require the bot to have the "View Audit Log" permission enabled at server level.
```

Ade2310 committed:
Hopefully the last one for now with this one

also, plooshcot does not canonically exust in bluey

MoonlightCapital reviewed (approved):
LGTM
